mindmeld/models/taggers/pytorch_crf.py

```diff
@@ -0,0 +1,412 @@
+import logging
+import os
+import random
+import uuid
+from collections import Counter
+from copy import copy
+from itertools import zip_longest
+from random import randint
+
+import numpy as np
+import torch
+import torch.nn as nn
+from sklearn.feature_extraction import DictVectorizer, FeatureHasher
+from sklearn.metrics import f1_score
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelEncoder
+from torch import optim
+from torch.utils.data import Dataset, DataLoader
+from torchcrf import CRF
+
+from ...exceptions import MindMeldError
+from ...path import USER_CONFIG_DIR
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PYTORCH_CRF_ER_CONFIG = {
+    "feat_type": "hash",  # ["hash", "dict"]
+    "feat_num": 50000,
+    "stratify": True,
+    "drop_input": 0.2,
+    "train_batch_size": 8,
+    "patience": 3,
+    "epochs": 100,
+    "train_dev_split": 0.15,
+    "optimizer_type": "sgd",  # ["sgd", "adam"]
+}
+
+
+class TaggerDataset(Dataset):
+    def __init__(self, inputs, seq_lens, labels=None):
+        self.inputs = inputs
+        self.labels = labels
+        self.seq_lens = seq_lens
+        self.max_seq_length = max(seq_lens)
+
+    def __len__(self):
+        return len(self.seq_lens)
+
+    def __getitem__(self, index):
+        mask_list = [1] * self.seq_lens[index] + [0] * (self.max_seq_length - self.seq_lens[index])
+
+        mask = torch.as_tensor(mask_list, dtype=torch.bool)
+        if self.labels:
+            return self.inputs[index], mask, self.labels[index]
+        else:
```

Comment:
you don't need `else` if you return early above :)

Reply:
Gotcha! Fixed!